src/InferArguments.cpp

```diff
@@ -0,0 +1,186 @@
+#include <set>
```

Comment:
Yes. It is needed in more than one place now. (I feel like we've moved this piece of functionality before, but perhaps this is just agents rewriting the matrix.)